lib/plugins/aws/package/compile/events/schedule.js

```diff
@@ -148,69 +191,120 @@ class AwsCompileScheduledEvents {
-              }
+              if (method === METHOD_SCHEDULER) {
+                const targetObj = {
+                  Arn: lambdaTarget,
+                  RoleArn: roleArn,
+                };
+                if (Input) {
+                  Object.assign(targetObj, { Input });
+                }
 
-              const scheduleTemplate = `
-              {
-                "Type": "AWS::Events::Rule",
-                ${dependsOn ? `"DependsOn": "${dependsOn}",` : ''}
-                "Properties": {
-                  "ScheduleExpression": ${templateScheduleExpression},
-                  "State": "${State}",
-                  ${Name ? `"Name": "${Name}",` : ''}
-                  ${Description ? `"Description": "${Description}",` : ''}
-                  "Targets": [{
-                    ${Input ? `"Input": "${Input}",` : ''}
-                    ${InputPath ? `"InputPath": "${InputPath}",` : ''}
-                    ${InputTransformer ? `"InputTransformer": ${InputTransformer},` : ''}
-                    "Arn": ${lambdaTarget},
-                    "Id": "${scheduleId}"
-                  }]
+                const properties = {
+                  ScheduleExpression,
+                  State,
+                  Target: targetObj,
+                  FlexibleTimeWindow: {
+                    Mode: 'OFF',
+                  },
+                };
+                if (Name) {
+                  Object.assign(properties, { Name });
                 }
-              }
-            `;
-
-              const permissionTemplate = `
-              {
-                "Type": "AWS::Lambda::Permission",
-                ${dependsOn ? `"DependsOn": "${dependsOn}",` : ''}
-                "Properties": {
-                  "FunctionName": ${lambdaTarget},
-                  "Action": "lambda:InvokeFunction",
-                  "Principal": "events.amazonaws.com",
-                  "SourceArn": { "Fn::GetAtt": ["${scheduleLogicalId}", "Arn"] }
+                if (Description) {
+                  Object.assign(properties, { Description });
+                }
+                if (timezone) {
+                  Object.assign(properties, { ScheduleExpressionTimezone: timezone });
+                }
+                if (dependsOn) {
+                  Object.assign(properties, { DependsOn: dependsOn });
                 }
-              }
-            `;
 
-              const newScheduleObject = {
-                [scheduleLogicalId]: JSON.parse(scheduleTemplate),
-              };
+                const schedule = {
+                  Type: 'AWS::Scheduler::Schedule',
+                  Properties: properties,
+                };
 
-              const newPermissionObject = {
-                [lambdaPermissionLogicalId]: JSON.parse(permissionTemplate),
-              };
+                const newScheduleObject = {
+                  [scheduleLogicalId]: schedule,
+                };
+
+                _.merge(
```

Comment:
Let's use `Object.assign` (ideally `_.merge` should be replaced in most places it's used, but ofc it's not scope of this PR)